casper/src/main/scala/coop/rchain/casper/BlockExecutionTracker.scala

```diff
@@ -0,0 +1,39 @@
+package coop.rchain.casper
+
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.syntax.all._
+import coop.rchain.blockstorage.dag.BlockDagStorage.DeployId
+import coop.rchain.rholang.interpreter.EvaluateResult
+import coop.rchain.sdk._
+
+trait BlockExecutionTracker[F[_]] {
+  def execStarted(d: DeployId): F[Unit]
+  def execComplete(d: DeployId, res: EvaluateResult): F[Unit]
+}
+
+sealed trait DeployStatus
+case object DeployStatusStarted                    extends DeployStatus
+final case class DeployStatusError(status: String) extends DeployStatus
+
+object StatefulExecutionTracker {
+  def apply[F[_]: Sync]: F[StatefulExecutionTracker[F]] =
+    for {
+      ref <- Ref.of(Map.empty[DeployId, DeployStatus])
+    } yield new StatefulExecutionTracker(ref)
```

Comment:
Why it is needed? The only reason in my opinion is not to write `new` when instantiating.

Reply:
There are two reasons:
1. make creation lazy so any computation in constructor will not be executed on call but when F value is executed
2. in this case creation of Ref requires F context